### ISSUE TITLE
Link framework build to NPM

### ISSUE THREAD

**Issue by kamerat** [logan-laughery/field-guide-ui#34]
We have to update the package.json file to support NPM.

This is needed so that when a user does `npm i field-guide-ui`, he will get the required files only (not docs for example) and also be able to use the framework via `import FieldGuideUI from 'field-guide-ui'`

### Resources
https://www.telerik.com/blogs/vuejs-how-to-build-your-first-package-publish-it-on-npm
https://vuejs.org/v2/cookbook/packaging-sfc-for-npm.html

**Comment by kamerat** [comment]
I have not done this myself before. I skimmed trough the two resources provided but did not understand the part with the "rollup". Essentially what file should be the entry to NPM. Should it be the output of the build or should it be [/src/components/index.js ](https://github.com/logan-laughery/field-guide-ui/blob/master/src/components/index.js ) 🤷‍♂️ 

Anyone have insight here?

**Comment by logan-laughery** [comment]
I haven't had a chance to read through the resources you posted yet but I know I've had success publishing single components to npm by following this medium post: https://medium.com/justfrontendthings/how-to-create-and-publish-your-own-vuejs-component-library-on-npm-using-vue-cli-28e60943eed3

It outlines how vue cli can be used to generate bundled files containing only the components we care about as well how to control what gets uploaded to npm.

**Comment by kamerat** [comment]
This looks perfect in regards to NPM, My only concern is that it forces you/the user to import the whole thing and it automagically globally installs itself. I'm more in to being able to choose what type of components I would like to use on each pages I create. This way it is way easier to refactor, knowing what is used where. Also, I think it will be more gentle on code splitting, but don't quote me on that.

This is what I tried to achieve in my pr #31. It works locally but not sure if it will work the way the article states. (I might think it does, but not sure)

**Comment by logan-laughery** [comment]
Yep, completely agree. I prefer the method for installation/usage you introduced in #31 over what is offered in the article.  I'll get that merged in.

Hopefully, the methods for building and setting up the files attribute of package.json presented in the post will work for our NPM set up.
